src/cascadia/TerminalApp/TabBase.cpp

```diff
@@ -542,6 +583,15 @@ namespace winrt::TerminalApp::implementation
             {
                 tabItemResources.Remove(key);
             }
+
+            for (const auto& [k, v] : tabItemThemeResources)
+            {
+                const auto& currentDictionary = v.as<ResourceDictionary>();
+                if (currentDictionary.HasKey(key))

```

Comment:
The `IDictionary` docs say that `Remove()` does not throw if the key does not exist. I'm assuming the same is true for WinRT's `IMap`. Is this extra check really needed?